src/System/Metrics/Protocol/Forwarder.hs

```diff
@@ -21,33 +22,32 @@ import           System.Metrics.Protocol.Type
 --
 data EKGForwarder req resp m a = EKGForwarder {
     -- | The acceptor sent us a request for new metrics.
-    recvMsgReq   :: req -> m resp
+    recvMsgReq  :: req -> m resp
 
     -- | The acceptor terminated. Here we have a pure return value, but we
     -- could have done another action in 'm' if we wanted to.
-  , recvMsgDone  :: m a
+  , recvMsgDone :: m a
   }
 
 -- | Interpret a particular action sequence into the server side of the
 -- 'EKGForward' protocol.
 --
 ekgForwarderPeer
-  :: Monad m
+  :: forall m req resp a. ()
+  => Monad m
   => EKGForwarder req resp m a
-  -> Peer (EKGForward req resp) 'AsServer 'StIdle m a
+  -> Server (EKGForward req resp) 'NonPipelined 'StIdle m a
 ekgForwarderPeer EKGForwarder{..} = go
   where
+    go :: Server (EKGForward req resp) 'NonPipelined 'StIdle m a
     go =
       -- In the 'StIdle' state the forwarder is awaiting a request message
       -- from the acceptor.
-      Await (ClientAgency TokIdle) $ \case
-        -- The acceptor sent us a request for new metrics, so now we're
-        -- in the 'StBusy' state which means it's the forwarder's turn to send
-        -- a reply.
-        MsgReq req -> Effect $ do
+      Await \case
+        MsgReq req -> Effect do
           resp <- recvMsgReq req
-          return $ Yield (ServerAgency TokBusy) (MsgResp resp) go
+          return do Yield (MsgResp resp) go
```

Comment:
Using `do` as `$` is rather extravagant :grin:.

Reply:
Yeah I didn't like it actually, I have gone too far this time!